src/snowflake/snowpark/modin/pandas/dataframe.py

```diff
@@ -152,19 +152,35 @@ def __init__(
         # Siblings are other dataframes that share the same query compiler. We
         # use this list to update inplace when there is a shallow copy.
         from snowflake.snowpark.modin.pandas.utils import try_convert_index_to_native
+        from snowflake.snowpark.modin.plugin.extensions.index import Index
 
         self._siblings = []
 
+        if isinstance(index, DataFrame):  # pandas raises the same error
+            raise ValueError("Index data must be 1-dimensional")
+
         # Engine.subscribe(_update_engine)
+        if isinstance(data, Index):
```

Comment:
you can add this block under line 180. For all lazy data cases, i.e., `if isinstance(data, (DataFrame, Series, Index))`, go to the first fork.

Reply:
Can you refactor this constructor to make it cleaner:

if all inputs (data, columns, index) are all local, create the native dataframe and then convert to lazy df.
otherwise, create the lazy df from data and then join extra lazy columns if given, lastly join the lazy index.